rfcs/0000-wifi-buddy-interface.md

```diff
@@ -0,0 +1,125 @@
+# RFC - ESP32 Wifi Buddy Interface Design
+
+## Goal
+
+To bring wireless networking to MnemOS Beepy devices, an [ESP32-C3] SoM will be
+integrated to serve as a WiFi coprocessor. This device will run MnemOS-based
+firmware, which we are referring to as "WiFi Buddy". See [#121] for details on
+the rationale for choosing the ESP32-C3 as the WiFi device, and for discussion
+of the overall design for ESP32-C3 integration.
+
+The goal of this document is to propose an overall design for the hardware and
+software interfaces between MnemOS on the CPU and MnemOS on the WiFi Buddy.
+
+## Background
+
+### Hardware
```

Comment:
Might be worth mentioning/discussing https://github.com/tosc-rs/mnemos/tree/main/hardware/esp32c3-wifi-dev, and particularly note that we'll be able to use:

* 4-pin SPI
* 2-pin UART
* 2 signal GPIOs (could be 4, if we don't use UART)

For initial development.